### PR TITLE
CL compiler compat

### PR DESCRIPTION
**Pull request by jschueller**
- Drop remaining use of vlas in tests
- Add CI tests for MS C compiler in combinations with intel fortran compilers.

### CODE REVIEW

**Comment by zaikunzhang** [comment]
Hi Julien @jschueller , thank you for the very prompt response, as always. Since this PR is only about tests and all of them pass, I can merge it when you think it is ready. Thanks!

File: c/examples/cobyla/cobyla_example.c [hunk 1]
```diff
@@ -23,7 +23,7 @@ int main(int argc, char * argv[])
   double x[2] = {0.0, 0.0};
   double f = 0.0;
   double cstrv = 0.0;
-  double nlconstr[m_nlcon];
+  double nlconstr[1]; // of size m_nlcon
```
Comment (zaikunzhang) [inline c/examples/cobyla/cobyla_example.c:26]: Just curious, will `nlconstr[]` work? `nlconstr[1]` seems a bit misleading. I have no idea. Thanks.

Comment (jschueller) [reply c/examples/cobyla/cobyla_example.c]: ok, lets use m_nlcon with a define

**Comment by jschueller** [comment]
it should be ok to merge when CI is green

**Pull request merged**